cmd/agent/app/builder.go

```diff
@@ -100,8 +98,11 @@ type HTTPServerConfiguration struct {
 }
 
 // WithReporter adds auxiliary reporters.
-func (b *Builder) WithReporter(r reporter.Reporter) *Builder {
-	b.otherReporters = append(b.otherReporters, r)
+func (b *Builder) WithReporter(r reporter.ReporterBuilder) *Builder {
```

Comment:
For multiple builders, I think this is meant to be called more than once, right?

```go
b.WithReporter(reporterBuilderA).WithReporter(reporterBuilderB)
```

Wouldn't it be better to accept multiple builders at once? Like:

```go
b.WithReporters(reporterBuilderA, reporterBuilderB)
```

Reply:
+1, can make it a vararg